### DISCUSSION TITLE
[Question] Can Fingerprint Identification detect user uniquely who is using kasm destroyable browser session ?

### DISCUSSION THREAD

**Discussion by Greatz08** [fingerprintjs/fingerprintjs#1049] [Q&A]
Kasm workspace give users option to create destroyable browsers sessions too like for 1,2 or more hrs and than total chromium or firefox session is deleted so can Fingerprint Identification (closed source, proprietary solution) detect those users uniquely who are using this for anonymous surfing ?

**Comment by Valve** [comment]
Hi @Greatz08 
Thanks for your question. 
We didn't test our Pro version on the Kasm workspace product, however our product has a free unlimited 14-day trial, which you can start on dashboard.fingerprint.com/signup
Please feel free to try it out and get back with your results. 
If you observe any misidentifications, such as false positive errors (two unrelated browsers having the same visitor ID) or false negative errors (one same browser has two different visitor IDs across two different sessions), please report those errors and we'll submit them to our product & engineering teams.

**Comment by Greatz08** [comment]
@Valve 

Free :
![image](https://github.com/user-attachments/assets/c90c6ca1-b50c-4c30-9bd1-d92b4af64498)

Pro :
![image](https://github.com/user-attachments/assets/7f1a6157-5c42-4a22-a65f-f44525a6b863)


Kasm firefox session was running which i will delete now :

![image](https://github.com/user-attachments/assets/e808a48d-40fd-4b17-b37d-3662daff3136)

 
After deleting firefox session completely from kasm and Recreating it from scratch Results were same with both free and pro versions so it is indeed able to catch  : 

![image](https://github.com/user-attachments/assets/26e9b6c1-2033-4718-aa8a-4f5dba12a501)


![image](https://github.com/user-attachments/assets/3a86554c-f0d5-423e-b9ec-0d508e90016c)


I did changed useragent to safari one and still it was able to detect because it observes way to many things so through ip and other stuff it was still able to detect uniquely 

![image](https://github.com/user-attachments/assets/a14932cf-2768-4ca7-a104-a0f2f0f985df)
 

But i believe IF there are multiple kasm workspaces running with same ip and same kasm version and same firefox and IT IS POSSIBLE EASILY as Many countries use CGNAT because of this same ip is shared between several 100's of users and running on same configurations is easily possible with docker so  if each and everyone is using same components  then it will get confused and show same unique identifier for everyone  because firefox itself is DEFAULT we did nothing so it will check for rest other system environments and if those will be same then it will show same unique identifier for 100's of user i guess 

And there is one more possibility which i tested in which you run a whole operating system in kasm workspace like debian,ubuntu etc and inside those os you install and use firefox as legit os system and then delete operating system session and when we will reinstall firefox it will show different unique identifier for same person.

But still still it is a very strong fingerprinting system i have to admit and max wont be using like this so it will  most probably detect unique visitors no matter what they use . Users of tails os or destructible os session or variations in ip and user agent with destructible session can only bypass its detection

**Comment by Valve** [comment]
I think there will be a massive difference in the false positive rate between FingerprintJS and Fingerprint Identification.
While the false negative rate is the same, there will be a higher probability for someone else to get the same visitor ID as you had on the free version.
